source/Octostache/Templates/Functions/TextManipulationFunction.cs

```diff
@@ -137,6 +138,96 @@ internal class TextManipulationFunction
             }
         }
 
+        public static string? Indent(string? argument, string[] options)
+        {
+            if (argument == null)
+            {
+                return null;
+            }
+
+            if (argument.Length == 0)
+            {
+                // No content, no indenting
+                return string.Empty;
```

Comment:
This might lead to unpredictable results if you were relying on a template to write itself in an indented format. For example something like `#{MyPrefix | Indent 2}#{MySuffix}` would not work if there were cases where `MyPrefix` was an optional variable.

Reply:
This decision was based on the behaviour of Python's [textwrap.indent](https://docs.python.org/3/library/textwrap.html#textwrap.indent) function which doesn't apply the initial indent to the empty string.

When working with indents it is generally better to indent as the last operation. So I would suggest the example above be written as:

```
#{MyPrefix | Append #{MySuffix} | Indent 2}
```

This makes it clearer that the suffix will be part of the indent (and will be indented if it has multiple lines).

Happy to change, just wanted to provide the context.

Reply:
👍

Reply:
I think is fine as long as we call this out in the docs, as I believe other template languages like Helm will apply the indent even to an empty string.

Reply:
Actually, on a second look, Helm doesn't indent empty strings either.